.github/actions/create-c3-docs-pr/action.yml

```diff
@@ -0,0 +1,98 @@
+name: "Create C3 Docs PR"
+description: "Creates a PR to Cloudflare Docs most recent C3 diffs for full-stack frameworks"
+inputs:
+  token:
+    description: "Github access token"
+    required: true
+  releaseBranch:
+    description: "Name of the release branch containing diff workflow run"
+    required: true
+  targetHeadBranch:
+    description: "The name of the branch to push to in the target repo"
+    required: true
+  targetBaseBranch:
+    description: "The name of the base (main) branch to PR against in the target repo"
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Lookup Workflow ID
+      id: lookup-workflow-id
+      uses: actions/github-script@v7
+      with:
+        retries: 3
+        retry-exempt-status-codes: 400,401,403
+        github-token: ${{ inputs.token }}
+        script: |
+          const {data: workflows} = await github.rest.actions.listWorkflowRuns({
+            owner: "cloudflare",
+            repo: "workers-sdk",
+            branch: "${{ inputs.releaseBranch }}",
+            workflow_id: "c3-e2e.yml",
+            per_page: 1,
+          });
+          return workflows.workflow_runs[0].id;
+
+    # For debug purposes
+    - name: Echo Workflow Id
+      shell: bash
+      run: |
+        echo ${{ steps.lookup-workflow-id.outputs.result }}
+
+    - name: Configure Git
+      shell: bash
+      run: |
+        git config --global user.email wrangler@cloudflare.com
+        git config --global user.name 'Wrangler automated PR updater'
+
+    - name: Checkout Destination Repo
+      uses: actions/checkout@v3
+      with:
+        repository: cloudflare/cloudflare-docs
+        path: ./clone-dir
+        ref: ${{ inputs.targetBaseBranch }}
+        token: ${{ inputs.token }}
+        fetch-depth: 0
+
+    - name: Create Branch
+      working-directory: ./clone-dir
+      shell: bash
+      run: |
+        git checkout -b "c3-diffs-${{ github.sha }}"
+
+    - name: Download Artifacts
+      uses: actions/download-artifact@v4
+      with:
+        name: e2e-framework-diffs
+        path: ./clone-dir/content/pages/framework-guides/c3-framework-diffs
+        github-token: ${{ inputs.token }}
+        run-id: ${{ steps.lookup-workflow-id.outputs.result }}
+
+    - name: Git Commit
+      id: git-commit
+      working-directory: ./clone-dir
+      shell: bash
+      run: |
+        git add diffs
+        if git status | grep -q "Changes to be committed"
+        then
+          git commit -m "Updating diffs"
+          git push -f -u origin HEAD:${{ inputs.targetHeadBranch }}
```

Comment:
I think it's safe for now since we're only using in one place. Should we decide to abstract this into a more general purpose "create-pr-in-another-repo" action that's used in multiple places, a check would make sense, and we'd end up doing it in the callsite, not in here.

Reply:
sure 🙂👍

Reply:
my 2 cents: you should validate the branch name if you're force pushing but it's your decision to make 👍